File: 11-container-with-most-water/11-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        left = 0
        right = len(height) - 1
        maxWater = 0
        while left < right:
            if height[left] < height[right]:
                maxWater = max(height[left]*(right-left), maxWater)
                left += 1
                #right += 1
            else:
                maxWater = max(height[right]*(right-left), maxWater)
                right -= 1
        return maxWater

File: 11-container-with-most-water/11-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
   let maxArea= 0;
   let left= 0;
   let right = height.length-1;
    // we will keep moving till we cross the right
    while(left<right){
        let width= right-left;
        let minHeight= Math.min(height[left],height[right]);
        maxArea = Math.max(maxArea,(width*minHeight));
        
        // the logic is we will keep going closer to the max height either from left or right
        if(height[left]<height[right]){
            left++;
        }else{
            right--;
        }
    }
    return maxArea;
};
